File: Orquestra.Domain/Entities/Inventory.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Orquestra.Domain.Entities;

public sealed class Inventory : Audit
{
    [Key]
    public Guid InventoryId { get; set; }

    public Guid CompanyId { get; set; }
    [ForeignKey(nameof(CompanyId))]
    public Company? Company { get; set; }

    [MaxLength(120)]
    public string Name { get; set; } = string.Empty;

    [MaxLength(255)]
    public string? Description { get; set; }

    public int Quantity { get; set; }
    public decimal? UnitPrice { get; set; }

    public byte[]? Image { get; set; }
    public string? ImageContentType { get; set; }

    [NotMapped]
    public decimal? TotalValue => UnitPrice.HasValue ? Quantity * UnitPrice.Value : null;
}

File: Orquestra.Frontend/app/api/consts/inventory.ts
import { BASE } from '@/app/api/fetch';
import { Guid } from 'guid-typescript';
import { iCompanyOutput } from './company';

const controller = 'api/Inventory';

export const CONSTS_INVENTORY = {
    post: `${BASE}/${controller}`,
    put: `${BASE}/${controller}`,
    getAllByCompanyId: `${BASE}/${controller}/GetAllByCompanyId`,
    disable: `${BASE}/${controller}/disable`
};

export interface iInventory {
    inventoryId?: Guid;
    companyId?: Guid;
    company?: iCompanyOutput | null;

    name?: string | null;
    description?: string | null;

    quantity?: number;
    unitPrice?: number | null;

    image?: number[] | null; // Entity;
    imageFormFile?: globalThis.File | null; // Input;
    imageBase64?: string; // Output;
    imageContentType?: string | null;

    totalValue?: number | null;
}

export interface iInventoryPaginated {
    output: iInventory[];
    count: number;
}